Scene ui improvements

WithoutPants opened the pull request:
This is a collection of UI improvements for the scene list page. The improvements come from #220 and #108 
Each improvement is done as a separate commit so that we can pick and choose which to include.

### Scene card overlays
I lifted the code from @echo6ix 's fork and tidied it up. It removes the filesize, duration and resolution from below the thumbnail, and overlays the resolution and duration to the bottom right of the thumbnail. See below:

![image](https://user-images.githubusercontent.com/53250216/69834361-f5198300-128d-11ea-94f6-33d768d47d0f.png)

### Handling portrait videos
I've been trying to get a consistent height shown for the scene videos when there is a mix of portrait and landscape videos. The CSS has proved a bit of a nightmare. Here is the result so far:

![image](https://user-images.githubusercontent.com/53250216/69834497-dbc50680-128e-11ea-971f-a47bf2ffc743.png)

The intent is to line up the thumbnails so that the title appears in the same place vertically. This appears to be working correctly, though I notice a bit of a performance hit, so I'm going to need to revisit it. Note also that the presence of the details field moves the divider down.

### Filter controls

I removed the text where possible. I'm not sold on how it looks - I think it needs further refinement.

![image](https://user-images.githubusercontent.com/53250216/69834586-67d72e00-128f-11ea-9a2f-e74238f5b1a0.png)

The filter and sort buttons I think are improved. The view buttons are ok. The select all/none buttons I think don't really look that good. I'm wondering if there is a better way to present those buttons. I don't know why the tooltip is so damn large.

# Scene performer thumbnails

Probably my favourite of the new things:

![image](https://user-images.githubusercontent.com/53250216/69834716-44f94980-1290-11ea-91bc-db45a6848d75.png)

Fairly self explanatory. 

Suggestions and feedback welcomed.

WithoutPants commented:
### Studio overlay

Added studio logo to the upper right corner of the scene card. The image is top-right aligned. The empty space in the right most thumbnail is due to empty space in the studio image.

![image](https://user-images.githubusercontent.com/53250216/69835437-cef7e100-1295-11ea-94e8-bd3b2a8731b8.png)

WithoutPants commented:
### Fade out scene overlays on hover

I can't show a screenshot of this as I don't have the required tools, but hovering over the scene card should fade out the overlays, and fade back in nicely when moving off the scene card.

Leopere commented:
This is exciting work!

StashAppDev commented:
Just checked out this branch.  The portrait display commit definitely needs work.  The fixed height of 240px is problematic.  The overlay shows up under the video rather than on top of it for a majority of my library.

I'm not sure why but it looks like the studio overlay isn't working either.  The opacity is always 1%, basically invisible.

WithoutPants commented:
Latest commit should fix the overlay positioning. I want to do some further work to try to align the position of the titles, as different video sizes make the titles in different vertical positions. I'm not able to reproduce the opacity issue so far.

echo6ix commented:
@WithoutPants, this looks good. I'm looking forward to trying it in an alpha_dev version.

I hit a wall on my end with my "compact grid view" mode, which took the scene card improvements a step further and displayed the string Studio name (using .bp3-text-overflow-ellipsis) on the thumbnail where you have the studio image. I personally think it's just a cleaner look since a lot of the studio images are meh, if they exist at all.

Would you consider adding the above as a [switch](https://blueprintjs.com/docs/#core/components/switch) option in settings. Where the switch is located at `Settings > Interface Config > Scene Cards > Display studio using logo {on/off toggle switch}`

StashAppDev commented on ui/v2/src/index.scss:
```diff
@@ -38,7 +38,7 @@ code {
   flex-flow: row wrap;
   justify-content: center;
   margin: $pt-grid-size $pt-grid-size 0 0;
-  padding: 0 calc(10%);
```
Can we make this and line 71 an interface option?

WithoutPants replied:
For line 71, I've added a zoom slider to adjust the sizes of the scene cards to one of four different sizes. These sizes can be overridden with the custom css if wanted.

For line 41, I think we should just make a reasonable default and allow it to be overridden with custom css.

StashAppDev commented:
Here's a diff which fixes the studio opacity issue, removes what seems to be unnecessary CSS, and fixes an issue I'm seeing where the performer images aren't showing:

```
diff --git a/ui/v2/src/index.scss b/ui/v2/src/index.scss
index b1ece78..b23d153 100755
--- a/ui/v2/src/index.scss
+++ b/ui/v2/src/index.scss
@@ -199,7 +199,7 @@ video.preview.portrait {
   font-weight: 400;
   width: 40%;
   height: 20%;
-  opacity: 75%;
+  opacity: .75;
   z-index: 9;
 }
 
@@ -294,19 +294,15 @@ span.block {
 }
 
 .performer-tag-container {
-  max-width: 100%;
-  height: 100%;
-  display: inline-block;
-  text-align: center;
   margin: 5px;
 }
 
 .performer-tag.image {
-  max-height: 150px;
-  max-width: 100%;
-  background-size: cover !important;
-  background-position: center !important;
-  background-repeat: no-repeat !important;
+  height: 150px;
+  width: 100%;
+  background-size: cover;
+  background-position: center;
+  background-repeat: no-repeat;
   margin: 0 auto;
 }
```

StashAppDev commented:
![group - Copy](https://user-images.githubusercontent.com/24867471/69903330-b565a080-134c-11ea-9191-fcca5f6c0229.jpg)

Another issue.  The button selection state is inverted.  It looks like it's due to the minimal class.

WithoutPants commented:
Thanks @StashAppDev . I've fixed the opacity and performer issues, and I've removed the `minimal` class from the view buttons. I've also fixed the overlay so that it shows the studio when the logo is hovered. I've also added a zoom slider to allow selecting the size of the scene cards.

Things left to do:
- [ ] add option to show studio by text. In future, we may want this to be a studio-specific option, but that will require a schema change.
- [ ] possibly condense view buttons further by changing to drop-down

WithoutPants commented:
Updated filter bar:

![image](https://user-images.githubusercontent.com/53250216/70185034-2c2fde80-173d-11ea-93d0-2fecfc4b7ca6.png)


Moved the select all/none buttons into a More button. Seemed a better fit in a separate more button, and I think it ends up looking cleaner.

I investigated changing the view buttons into a drop-down list, but the result doesn't save any space and adds an extra click to change views. The intent is a little more clear on the drop-down list, but I think ultimately having three buttons is better. Below is a comparison of the two options:

![image](https://user-images.githubusercontent.com/53250216/70185165-64372180-173d-11ea-9140-35d015a23c7b.png)

WithoutPants commented:
Also added option to display studios as text instead of logos, per @echo6ix 's request. The result of setting this option is shown below. The text highlights blue on mouseover.

![image](https://user-images.githubusercontent.com/53250216/70185367-bf691400-173d-11ea-9991-b924088d4f28.png)

WithoutPants commented:
For now I'm satisfied with these improvements, but I realise that UI design is highly opinionated. I hope I've addressed all of the concerns raised.

bnkai commented:
I've tested a little all of the above and they seem ok now.I especially like the slider for the screenshot size and the performer thumbnails.
A little thing i noticed is the overlay on the portrait mode pics. 
![overlay](https://user-images.githubusercontent.com/48220860/70245774-52766e00-177f-11ea-8443-9ad0ed26cf54.png)
